src/nbl/video/CSurfaceVulkan.cpp

```diff
@@ -0,0 +1,70 @@
+#include "nbl/video/surface/CSurfaceVulkan.h"
+
+#include "nbl/video/CVulkanCommon.h"
+#include "nbl/video/CVulkanPhysicalDevice.h"
+#include "nbl/video/CVulkanConnection.h"
+
+namespace nbl::video
+{
+
+template <typename Window>
+core::smart_refctd_ptr<CSurfaceVulkan<Window>> CSurfaceVulkan<Window>::create(
+    core::smart_refctd_ptr<video::CVulkanConnection>&& api,
+    core::smart_refctd_ptr<Window>&& window)
+{
+    if (!api || !window)
+        return nullptr;
+
+    if constexpr (std::is_same_v<Window, ui::IWindowWin32>)
+    {
+        VkWin32SurfaceCreateInfoKHR createInfo = { VK_STRUCTURE_TYPE_WIN32_SURFACE_CREATE_INFO_KHR };
+        createInfo.pNext = nullptr; // pNext must be NULL
+        createInfo.flags = static_cast<VkWin32SurfaceCreateFlagsKHR>(0);
+        createInfo.hinstance = GetModuleHandle(NULL);
+        createInfo.hwnd = (static_cast<HWND>(window->getNativeHandle()));
+
+        VkSurfaceKHR vk_surface;
+        if (vkCreateWin32SurfaceKHR(api->getInternalObject(), &createInfo, nullptr,
+            &vk_surface) == VK_SUCCESS)
+        {
+            return core::make_smart_refctd_ptr<this_t>(std::move(api), std::move(window), vk_surface);
```

Comment:
need 2 lines and a protected constructor

Reply:
@achalpandeyy status? if untouched, tag Erfan

Reply:
@Erfan-Ahmadi

Reply:
done:
https://github.com/Devsh-Graphics-Programming/Nabla/commit/894c8ac3943ba9ae5b7bf338e00592970d1020d7